neqo-qpack/src/encoder.rs

```diff
@@ -257,28 +271,42 @@ impl QPackEncoder {
         }
     }
 
-    fn change_capacity(&mut self, value: u64) -> Res<()> {
+    fn change_capacity(&mut self, value: u64) {
         qdebug!([self], "change capacity: {}", value);
-        self.table.set_capacity(value)?;
-        EncoderInstruction::Capacity { value }.marshal(&mut self.send_buf, self.use_huffman);
+        self.change_capacity = Some(value);
+    }
+
+    fn maybe_send_change_capacity(&mut self, conn: &mut Connection, stream_id: u64) -> Res<()> {
+        if let Some(cap) = self.change_capacity {
+            let mut buf = QPData::default();
+            EncoderInstruction::Capacity { value: cap }.marshal(&mut buf, self.use_huffman);
+            if !conn.stream_send_atomic(stream_id, &buf)? {
+                return Err(Error::EncoderStreamBlocked);
+            }
+            self.table.set_capacity(cap)?;
+            self.change_capacity = None;
+        }
         Ok(())
     }
 
     /// Sends any qpack encoder instructions.
     /// # Errors
     ///   returns `EncoderStream` in case of an error.
     pub fn send(&mut self, conn: &mut Connection) -> Res<()> {
-        if self.send_buf.is_empty() {
-            Ok(())
-        } else if let Some(stream_id) = self.local_stream_id {
-            let r = conn
-                .stream_send(stream_id, &self.send_buf[..])
-                .map_err(|_| Error::EncoderStream)?;
-            qdebug!([self], "{} bytes sent.", r);
-            self.send_buf.read(r as usize);
-            Ok(())
-        } else {
-            Ok(())
+        match self.local_stream {
+            LocalStreamState::NoStream => Ok(()),
```

Comment:
Just wanted to check this should be `Ok(())`. Is calling `send()` allowed when there isn't even an uninitialized stream? Does this deserve to be an error or emit a log message?